lib/proxy/helpers.rb

```diff
@@ -25,7 +25,8 @@ def log_halt code=nil, exception=nil
 
   # read the HTTPS client certificate from the environment and extract its CN
   def https_cert_cn
-    certificate_raw = request.env['SSL_CLIENT_CERT'].to_s
+    ssl_client_cert_header = Proxy::SETTINGS.ssl_client_cert_header || 'SSL_CLIENT_CERT'
+    certificate_raw = request.env[ssl_client_cert_header].to_s.gsub(/   /, "\n")
```

Comment:
Ok, I thought about it during weekend and I worked out something like this:
```
def strip_header_continuation_from_ssl_cert(ssl_client_cert)
  cert = "-----BEGIN CERTIFICATE-----\n"
  #skip header and footer of certificate from splitting with [2..-3]
  cert += ssl_client_cert.split(/[ |\t]+/)[2..-3].join("\n")
  cert += "\n-----END CERTIFICATE-----"
  cert
end
```
it should be valid for RFC, quite readable and testable.